src/libraries/System.Private.CoreLib/src/System/Environment.Windows.cs

```diff
@@ -122,6 +122,28 @@ public static string SystemDirectory
             }
         }
 
+        public static bool UserInteractive
+        {
+            get
+            {
+                // Per documentation of GetProcessWindowStation, this handle should not be closed
+                IntPtr handle = Interop.User32.GetProcessWindowStation();
+                if (handle == IntPtr.Zero)
+                {
+                    return false; // eg., Windows Nano
+                }
+
+                Interop.User32.USEROBJECTFLAGS flags = default;
+                if (!Interop.User32.GetUserObjectInformationW(handle, Interop.User32.UOI_FLAGS, ref flags, Marshal.SizeOf(flags), ref _))
```

Comment:
Please do not use `Marshal.SizeOf`. It is slow expensive API. Use `sizeof` instead.

See https://github.com/dotnet/runtime/blob/master/docs/coding-guidelines/interop-guidelines.md#definition

Reply:
OK. Why might I want to ever use `Marshal.SizeOf` in this context - and if there are no reasons, perhaps we should have an analyzer?

I read through https://github.com/dotnet/csharplang/issues/1828 in the hope it would explain this, but it only suggests why `Marshal.SizeOf` is not always functionally equivalent.

Reply:
Yes, the analyzer for this is listed in https://github.com/dotnet/corefx/issues/40739.

The only situation where `Marshal.SizeOf`, `Marshal.PtrToStructure` and friends make sense are non-blittable types. We try hard to avoid non-blittable structs for interop.